airflow/models/taskinstance.py

```diff
@@ -1601,7 +1596,8 @@ def _register_dataset_changes(self, *, session: Session) -> None:
                     session=session,
                 )
 
-    def _execute_task_with_callbacks(self, context, test_mode=False):
+    @provide_session
+    def _execute_task_with_callbacks(self, context, test_mode: bool = False, session: Session = NEW_SESSION):
```

Comment:
```suggestion
    def _execute_task_with_callbacks(self, context, test_mode: bool = False, *, session: Session):
```

This is only ever called with an explicit session, no point decorating.

Reply:
Fixed.